### ISSUE TITLE
Setup.py needs to pin pyparsing 2.4.7

### ISSUE THREAD

**Issue by daddycocoaman** [facebookincubator/python-nubia#85]
Hi,

I just spent the week troubleshoot a big problem for my tool based on Nubia and I have **finally** discovered the issue. 

Nubia has a dependency on `pyparsing>=2.2.0,<3` as defined in the requirements.txt, but neither the setup.py nor the pyproject.toml reflect this.  I use Poetry as my dev environment and it has a great dependency resolving system. When developing my tool, everything works great. However, when I try to `pip install` my tool to test it, **subcommand arguments do not work**. I dug through Nubia and found the location of the issue starts here:

https://github.com/facebookincubator/python-nubia/blob/9cb98e220a6fb7e837631f5b0f75a8b1363e5a0c/nubia/internal/parser.py#L47

It seems like there's been some change in pyparsing from 2.X to 3.X. Here is what happens when I run the command `dump heap filter=[Azure]` with each version. On the left is a working example using 2.4.7, and on the right is pyparsing 3.0.3. 

![image](https://user-images.githubusercontent.com/21189155/139511012-79a4e165-6790-4a63-8b6e-45e8bd441869.png)

Because the `identifier` is different, subcommands get parsed as normal commands when pyparsing 3.X is used. 

Pip doesn't know that the requirement for Nubia is `pyparsing>=2.2.0,<3` because pip has changed drastically over the years and requirements.txt doesn't really hold much of a priority anymore. At a minimum, setup.py needs to reflect this version limit anyway so the dev using Nubia doesn't install `pyparsing >=3`. Ideally, the parsing should be rewritten to handle 3.0 but that is probably a little more involved. 

It would be ideal to just move to Poetry since pyproject.toml is generally easier to handle than setup.py but that's up to you. It was requested here (https://github.com/facebookincubator/python-nubia/pull/21) previously.

TL;DR - Please pin pyparsing version in setup.py or upgrade version.

**Comment by daddycocoaman** [comment]
It should be noted that pyparsing 3.0 was [released only a few days ago](https://github.com/pyparsing/pyparsing/releases/tag/pyparsing_3.0.0) so maybe Nubia users haven't noticed yet but this will definitely be an upcoming issue, if not already.

**Comment by ddutt** [comment]
I ran into this problem too
